src/shared/ethtool-link-mode.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import re
+import shlex
+import subprocess
+import sys
+
+OVERRIDES = {
+    'autoneg' : 'autonegotiation',
+}
+
+mode, cpp, header = sys.argv[1:]
+xml = mode == '--xml'
+
+command = [*shlex.split(cpp), '-include', header, '-']
+out = subprocess.check_output(command, stdin=subprocess.DEVNULL, text=True)
```

Comment:
This makes C8S unhappy, since `text=` is a python 3.7 thing, but I completely missed https://github.com/systemd/systemd/commit/7c0d79131b136cb8dd63f0847844be2d79f6994c where we bumped the baseline to python 3.7, so I'll go ahead and bump the python in the C8S to 3.8 to fix this.

Reply:
I'll rework the `text=` attribute for compatibility. (We have `text=` in ukify, but ukify has stronger dependencies.)